### PR TITLE
Add a Jenkinsfile

### PR DESCRIPTION
**Pull request by eatyourgreens**
On merges to master, build a docker image called `zooniverse/zoo-ml-subject-assistant:${GIT_COMMIT}`
Update the kubernetes template to deploy that image.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
I copied this, more or less, from [zooniverse/front-end-monorepo](https://github.com/zooniverse/front-end-monorepo). I think the git commit hash is in the image name to ensure that we build a new image when code changes, rather than using an old image.

**Comment by camallen** [comment]
> I copied this, more or less, from [zooniverse/front-end-monorepo](https://github.com/zooniverse/front-end-monorepo). I think the git commit hash is in the image name to ensure that we build a new image when code changes, rather than using an old image.

Yep, this ensures Kubernetes detects replaces pods with the new image. If you use a normal docker `image_name:tag` e.g `zooniverse/zoo-ml-subject-assistant:production` or similar then K8 will not update the pods as it checks what tag was last deployed (`production`) and thus doesn't detect a change, so skips triggering deployment.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          }
+        }
+      }
+
+      post {
+        unsuccessful {
+          slackSend (
+            color: '#FF0000',
+            message: "DEPLOY FAILED: Job '${JOB_NAME} [${BUILD_NUMBER}]' (${BUILD_URL})",
+            channel: "#frontend-rewrite"
```
Comment (shaunanoordin) [inline Jenkinsfile]: This definitely needs to be removed, or perhaps sent to a different channel.

Comment (eatyourgreens) [reply Jenkinsfile]: That channel might be interested to know when we mess up.

**Comment by shaunanoordin** [comment]
Noting the lessons we learned with Docker (much thanks to @camallen ):

- The `Dockerfile` should NOT have...
```
ENTRYPOINT "node"
CMD ["server/proxy-server.js"]

```
- ...but should instead have...
```
CMD ["node", "server/proxy-server.js"]
```
- We could not detect that `Dockerfile` was breaking on `localhost`, because on local, `docker-compose` was _overwriting_ the borked launch from Dockerfile _with a launch that works:_
```
// docker-compose.yaml
services:
  proxy:
    ...
    entrypoint:
      - "npm"
    command: ["start"]
    ...
```

Basically, Dockerfile's Entrypoint !== docker-compose's Entrypoint.

Additional notes to self:
- All `zooniverse` repos are configured so that _any_ branch with a Jenkinsfile will be monitored. It's up to the Jenkinsfile to explicitly say "only run me if the branch is 'master'"
- `Dockerfile` runs on the Kubernetes server and on local, but `docker-compose.yaml` is only relevant to local.

**Pull request merged**